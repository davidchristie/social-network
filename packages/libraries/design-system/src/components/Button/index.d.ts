import React from "react";
import "./index.css";
interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> {
}
export default class Button extends React.Component<Props> {
    render(): JSX.Element;
}
export {};
