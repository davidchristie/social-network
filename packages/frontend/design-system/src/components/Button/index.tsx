import MaterialButton from "@material-ui/core/Button";
import React from "react";

interface Props {
  children?: React.ReactNode;
  className?: string;
  onClick?: (event: React.MouseEvent<HTMLElement>) => void;
  style?: "primary" | "secondary";
  type?: string;
  variant?: "contained" | "extendedFab" | "fab" | "flat" | "outlined" | "raised" | "text";
}

const Button: React.SFC<Props> = ({
  children,
  className,
  onClick,
  style,
  type,
  variant,
}) => {
  return (
    <MaterialButton
      className={className}
      color={style}
      onClick={onClick}
      type={type}
      variant={variant}
    >
      {children}
    </MaterialButton>
  );
};

export default Button;
