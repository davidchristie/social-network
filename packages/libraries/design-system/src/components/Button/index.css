.Button {
  align-items: center;
  background-color: white;
  border: solid black 1px;
  display: inline-flex;
  font-size: 1.1em;
  opacity: 0.6;
  transition: opacity 200ms ease-in-out;
}

.Button:hover {
  opacity: 1;
}
