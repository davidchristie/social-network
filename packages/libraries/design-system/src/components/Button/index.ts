import React from "react";

import elementWithClass from "../../utilities/elementWithClass";
import "./index.css";

interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> { }

const Button = elementWithClass<Props>("Button", "button");

export default Button;
