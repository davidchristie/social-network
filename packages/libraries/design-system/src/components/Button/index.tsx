import React from "react";

import elementWithClass from "../../utilities/elementWithClass";
import "./index.css";

interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> { }

export default elementWithClass<Props>("Button", "button");
