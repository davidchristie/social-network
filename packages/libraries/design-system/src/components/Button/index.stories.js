import React from "react";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";
import { action } from "@storybook/addon-actions";

import Button from ".";

const stories = storiesOf("Components", module);

stories.add(
  "Button",
  withInfo({ inline: true })(() => <Button
    children="Button"
    onClick={action("onClick")}
  />)
);
