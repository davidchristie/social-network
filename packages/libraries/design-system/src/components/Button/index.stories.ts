import Button from ".";
import addComponentStory from "../../utilities/stories/addComponentStory";

addComponentStory(Button);
