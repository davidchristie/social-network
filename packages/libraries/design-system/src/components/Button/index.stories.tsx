import { action } from "@storybook/addon-actions";
import { withInfo } from "@storybook/addon-info";
import { storiesOf } from "@storybook/react";
import React from "react";

import Button from ".";

const stories = storiesOf("Components", module);

stories.add(
  "Button",
  withInfo({ inline: true })(() => <Button
    children="Button"
    onClick={action("onClick")}
  />)
);
