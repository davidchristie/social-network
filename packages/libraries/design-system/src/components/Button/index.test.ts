import {
  describeWithCustomClass,
  describeWithNoProps
} from "test-utilities";

import Button from "./index";

describe("Button component", () => {
  describeWithNoProps(Button);

  describeWithCustomClass(Button);
});
