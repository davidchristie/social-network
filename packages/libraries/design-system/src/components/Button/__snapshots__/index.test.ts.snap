// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component with custom class matches snapshot 1`] = `
<button
  className="Button CUSTOM_CLASS"
/>
`;

exports[`Button component with no props matches snapshot 1`] = `
<button
  className="Button"
/>
`;
